README.md

```diff
@@ -65,10 +67,38 @@ To add a new handler, you can add a new file under `app/handlers/` and start add
 
 **Note:** The regex supports [named capture](https://www.regular-expressions.info/named.html). In this example, we have a name example of `sentence`. When the handler command will be called, a key in the parameter hash will be added: `command_arguments`. This key will point to a hash of the capture name and value. In this case, `command_arguments => {sentence: "the sentence you wrote"}`
 
+### Handling messages with defined parameters
+
+The regular expression matching in the previous example provides a mechanism to supply named parameters to a method. However, many handlers may want to use named & typed parameters. This can be done using the `base_command` and `parameters` options.
+
+We can rewrite the command above to
+
+```yaml
+- repeat_handler:
+    commands:
+      - name: Repeat
+        description: "`repeat [sentence]`: Repeats the sentence you wrote"
+        base_command: "repeat"
+        parameters:
+          - sentence: string
+          - times: 10
+        action: repeat
+```
+
+And call the command with
+
+`/slackify repeat sentence="Why hullo there" times=10`
```

Comment:
here this would work with a plaintext message. I'd remove the mention of the slash command to make it more generic